pkg/server/skydb/recordutil/recordutil.go

```diff
@@ -328,21 +328,33 @@ func RecordSaveHandler(req *RecordModifyRequest, resp *RecordModifyResponse) sky
 	makeAssetsCompleteAndInjectSigner(db, req.Conn, records, req.AssetStore)
 
 	// execute after save hooks
-	if req.HookRegistry != nil {
-		records = executeRecordFunc(records, resp.ErrMap, func(record *skydb.Record) (err skyerr.Error) {
-			originalRecord, _ := originalRecordMap[record.ID]
-			err = req.HookRegistry.ExecuteHooks(req.Context, hook.AfterSave, record, originalRecord)
-			if err != nil {
+	records = hookTriggerer.triggerIfRegistered(records, hook.AfterSave, true)
+
+	resp.SavedRecords = records
+	resp.SchemaUpdated = schemaExtended
+
+	return nil
+}
+
+type saveHookTriggerer struct {
+	Req               *RecordModifyRequest
+	Resp              *RecordModifyResponse
+	OriginalRecordMap map[skydb.RecordID]*skydb.Record
+}
+
+func (t saveHookTriggerer) triggerIfRegistered(records []*skydb.Record, kind hook.Kind, shouldLogErr bool) []*skydb.Record {
+	if t.Req.HookRegistry != nil {
+		return executeRecordFunc(records, t.Resp.ErrMap, func(record *skydb.Record) (err skyerr.Error) {
+			originalRecord, _ := t.OriginalRecordMap[record.ID]
+			err = t.Req.HookRegistry.ExecuteHooks(t.Req.Context, kind, record, originalRecord)
+			if shouldLogErr && err != nil {
```

Comment:
I'm just refactoring the code and keeping the behaviour unchanged. If `shouldLogErr` is in the struct, there would be two separate `saveHookTriggerer` in the record save handler. I don't know which one is better.